test/storage/index_manager_test.cpp

```diff
@@ -0,0 +1,117 @@
+#include "storage/index/index_manager.h"
+#include <string>
+#include <vector>
+#include "catalog/catalog_sql_table.h"
+#include "storage/index/index_factory.h"
+#include "util/test_harness.h"
+#include "util/transaction_test_util.h"
+
+namespace terrier::storage::index {
+struct IndexManagerTest : public TerrierTest {
+  void SetUp() override {
+    TerrierTest::SetUp();
+    txn_manager_ = new transaction::TransactionManager(&buffer_pool_, true, LOGGING_DISABLED);
+    index_manager_ = new IndexManager();
+  }
+
+  void TearDown() override {
+    TerrierTest::TearDown();
+    delete index_manager_;
+    delete txn_manager_;
+  }
+  storage::RecordBufferSegmentPool buffer_pool_{100, 100};
+  transaction::TransactionManager *txn_manager_;
+  IndexManager *index_manager_;
+};
+
+// Check the correctness of drop index
+// NOLINTNEXTLINE
+TEST_F(IndexManagerTest, DropIndexCorrectnessTest) {
+  auto txn0 = txn_manager_->BeginTransaction();
```

Comment:
[Major] Since the current implementation follows the non-blocking manner, a concurrent correctness test may be necessary.

Reply:
Thanks for your comments. We will add it later.